plugins/modules/api.py

```diff
@@ -469,6 +514,63 @@ def api_query(self):
         except LibRouterosError as e:
             self.errors(e)
 
+    def build_api_extended_query(self, item):
+        if item['attribute'] not in self.extended_query['attributes']:
+            self.errors("'%s' attribute is not in attributes:%s"
+                        % (item, self.extended_query['attributes']))
+        if item['is'] == 'eq' or item['is'] == '==':
+            return (self.query_keys[item['attribute']] == item['value'],)
+        elif item['is'] == 'not' or item['is'] == '!=':
+            return (self.query_keys[item['attribute']] != item['value'],)
+        elif item['is'] == 'less' or item['is'] == '<':
+            return (self.query_keys[item['attribute']] < item['value'],)
+        elif item['is'] == 'more' or item['is'] == '>':
+            return (self.query_keys[item['attribute']] > item['value'],)
+        elif item['is'] == 'in':
+            return (self.query_keys[item['attribute']].In(*item['value']),)
+        else:
+            self.errors("'%s' is not operator for 'is'"
+                        % item['is'])
+
+    def api_extended_query(self):
+        self.query_keys = {}
+        for k in self.extended_query['attributes']:
+            if k == 'id':
+                self.errors("'extended_query':'attributes':'%s' must be '.id'" % k)
+            self.query_keys[k] = Key(k)
+        try:
+            if self.extended_query['where']:
+                where_args = False
+                for i in self.extended_query['where']:
+                    if "or" in i.keys():
```

Comment:
This is equivalent (and easier to read IMO):
```suggestion
                    if "or" in i:
```